### PR TITLE
Update CCDB secret rotation instructions.

### LINKED ISSUE #744: [CAP-1083] CCDB encryption key rotation
https://github.com/cloudfoundry-incubator/kubecf/blob/master/doc/secret_rotation.md

### PR DESCRIPTION
**Pull request by KucharczykL**
Closes #744.

### CODE REVIEW

**Comment by KucharczykL** [comment]
I am a bit fuzzy on how the helm upgrade should work. AFAIK, you'd create YAML file and then do `helm upgrade -f secret-rotation.yaml` but then wouldn't that mean the file should include a list of all the existing encryption keys? Is it safe to assume in the example that the one and only existing key is going to be named `encryption_key_0`? If not, should we add instructions for finding this information out?

File: xml/cap_admin_ccdb_key_rotation.xml [hunk 1]
```diff
@@ -62,48 +67,24 @@ secrets:
...
-cd /var/vcap/packages/cloud_controller_ng/cloud_controller_ng; \
-bundle exec rake rotate_cc_database_key:perform'
+<screen>
+&prompt.user;helm upgrade kubecf
      </screen>
     </step>
     <step>
      <para>
-      Restart the <literal>api-group</literal> pod.
+      After Helm finishes its updates, trigger the
```
Comment (btat) [inline xml/cap_admin_ccdb_key_rotation.xml]:
```suggestion
      After &helm; finishes its updates, trigger the
```

File: xml/cap_admin_ccdb_key_rotation.xml [hunk 2]
```diff
@@ -62,48 +67,24 @@ secrets:
...
+      <literal>rotate-cc-database-key</literal> errand:
      </para>
-<screen>&prompt.user;kubectl delete pod api-group-0 --namespace scf --force --grace-period=0</screen>
+<screen>&prompt.user;kubectl patch qjob kubecf-rotate-cc-database-key \
+--namespace kubecf \
```
Comment (btat) [inline xml/cap_admin_ccdb_key_rotation.xml]: `replaceable` tags

**Comment by btat** [comment]
> 
> 
> I am a bit fuzzy on how the helm upgrade should work. AFAIK, you'd create YAML file and then do `helm upgrade -f secret-rotation.yaml` but then wouldn't that mean the file should include a list of all the existing encryption keys? Is it safe to assume in the example that the one and only existing key is going to be named `encryption_key_0`? If not, should we add instructions for finding this information out?

^^ @f0rmiga @andreas-kupries see question from @KucharczykL

**Comment by KucharczykL** [comment]
So I've looked at https://github.com/cloudfoundry-incubator/kubecf/blob/master/deploy/helm/kubecf/values.yaml and If I understand it correctly, the first encryption key is specified here so it shouldn't need to be specified again in the updated YAML file?

**Comment by f0rmiga** [comment]
@KucharczykL I'll answer your question (hopefully) and also point out another issue with the instructions proposed in this doc update.

#### About your question

The process of rotating a key involves:
1. Decrypting the current contents of the database - this is done using the old key, therefore you need to have it in hands.
2. Encrypting the same contents of the database with the new key, specified in the `current` key label.

You are correct that we have the defaults here: https://github.com/cloudfoundry-incubator/kubecf/blob/dd25a5e118fb48abc6f233f66e46eabd1922d44f/deploy/helm/kubecf/values.yaml#L314-L321. We should probably be more specific about these defaults. I.e. point the operator to these default values to be able to go from there in the first rotation.

#### The other issue

You cannot simply call `helm upgrade` and pass a YAML file with only the `ccdb` properties. We most likely also want to suggest `--reuse-values`.

**Comment by KucharczykL** [comment]
@f0rmiga Thank you for the reply!

> 1. Decrypting the current contents of the database - this is done using the old key, therefore you need to have it in hands.
> 2. Encrypting the same contents of the database with the new key, specified in the `current` key label.

I have added this to clarify further.

> You cannot simply call `helm upgrade` and pass a YAML file with only the `ccdb` properties. We most likely also want to suggest `--reuse-values`.

I have modified the line to `helm upgrade kubecf --values rotate-secret.yaml --reuse-values`. Is that OK?

**Comment by f0rmiga** [comment]
@KucharczykL Sorry, I missed your previous message. Yes, that's correct.

**Comment by f0rmiga** [comment]
For completeness, we need to add the instruction for the user to check:

```
helm get values --namespace kubecf kubecf
```

This will return the current `ccdb.encryption.rotation` values to be updated.

**Pull request merged**